proposals/infra/INF-0004-validator-hashing.md

```diff
@@ -127,38 +128,108 @@ apply the hash. Instead it should apply the `PREVIEW` sentinel value from the
+in the debug layer and runtime.
 
 For shaders with the `PREVIEW` hash, the shader is allowed to execute only if
 developer mode and the experimental feature D3D12ExperimentalShaderModels is
 enabled, otherwise the runtime will produce an error.
 
-For shaders with the `BYPASS` hash, the new behavior will allow shaders to run
-without validating the hash regardless of whether the machine is in developer
-mode as long as the shader targets a supported non-experimental shader model.
+For shaders with the `BYPASS` hash, shaders can run without validating the hash 
+regardless of whether the machine is in developer mode as long as the shader 
+targets a supported non-experimental shader model.
 
-All other values of the hash will be assumed to be real hash values.
+All other values of the hash are assumed to be real hash values.
 
 For shaders that contain a real hash value, the behavior is unchanged. The hash
-will be verified and the shader will execute as it does with existing versions
-of the runtime.
+is verified and the shader executes as it does with existing versions of the 
+runtime.
 
 > Note: an important behavior change here is that shaders with zero'd hash data
 > will always be treated as invalid by the runtime and rejected.
+> 
+> An exception is for a period of time already existing preview shaders that used 
+> zero'd hash data will be treated as equivalent to having the `PREVIEW` hash 
+> until those preview shaders are no longer supported in new preview runtimes.
 
-There will be support for DXIL validation in the debug layer when dxil.dll is
-present, printing any errors found, but letting the shader through. This
-validation likely will default to being enabled for shaders that use the
-`BYPASS` hash. Shaders with the `PREVIEW` hash will default to validation off,
-since the DXIL validator will be assumed to not be final. The defaults will be
-able to be overridden by the app.
+#### D3D Runtime Validation Control
 
-Similarly the runtime could also support performing DXIL validation that fails
-shader creation if an error is found. It would support similar app controls as
-the debug layer, except defaulting to validation off.
+Apps can configure how the runtime validates bytecode passed to it. 
+
+The validation referred to here is implementation exposed by `dxcompiler.dll` 
+or `dxil.dll`.  This is the same validation used when the compiler
+endorses bytecode at compile time by applying a hash.  So it would typically be 
+redundant to validate again, but it can be useful to validate shaders that have 
+the `BYPASS` hash, or have the option to force validation. 
+
+If there is bytecode to validate, the runtime and/or debug layer attempt to 
+load one of these dlls, `dxcompiler.dll` first. If either is available, 
+bytecode validation is possible. 
+
+The following flags in `d3d12.h` show the validation control options.  
+`ID3D12Device` exposes `ID3D12BytecodeOptions` to set the flags, shown 
+further below.
```

Comment:
I feel like the API details should be placed under a new `## Detailed Design` section. immediately preceding the `## Appendix 1: DXIL Hashing Algorithm` section.  In this section, the need for, and expected functionality, of the proposed interface could be discussed.